### ISSUE TITLE
22.1.2: UI sluggish, CPU frequently above 100%

### ISSUE THREAD

**Issue by 2019-05-10** [dbeaver/dbeaver#17068]
#### System information: 
- Operating system (distribution) and version Dbeian/SID
- DBeaver version 22.1.2.202207091909
- Additional extensions

#### Connection specification:
- Database name and version: Postgres 12
- Driver name org.postgresql:postgresql:RELEASE 42.2.5
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
Since update to 22.1.2 (no other changes, in particular "Result-Set fetch size" is unchanged!) DBeaver is sluggish.
Opening a table in editor takes several seconds, UI blocks completely until several Minutes after the table has opened.
CPU frequently jumps over 100% on a lot of operations (example: "Help -> Check for updates" takes several minutes and causes all the time 106-109% CPU load)

#### Steps to reproduce, if exist:
Just work as usual, open tables, query, check for updates

#### Include any warning/errors/backtraces from the logs
Nothing related

**Comment by E1izabeth** [comment]
Hello, @2019-05-10! Please, send the [log files](https://github.com/dbeaver/dbeaver/wiki/Log-files).

By the way, taking into account your issue https://github.com/dbeaver/dbeaver/issues/17071, I'd suggest to [reset workspace](https://github.com/dbeaver/dbeaver/wiki/Reset-workspace).

**Comment by ar-jan** [comment]
I updated from 22.1.0 to 21.1.2 and have the same issue. I have a large materialized view (PostgreSQL 14) which loads in a fraction of a second in 22.1.0, but around 15 seconds in 22.1.2. UI hangs for a while longer. Also everything feels sluggish when starting up. Downgrading resolves the problem, so I assume the workspace is fine.

The `.log` contains one entry that might be relevant: `!ENTRY org.jkiss.dbeaver.model 4 0 2022-07-13 14:39:27.627
!MESSAGE Cannot determine transaction isolation level due to connection hanging. Setting to NONE.`

**Comment by ShadelessFox** [comment]
Hello @ar-jan,

Unfortunately, this one entry is not really relevant. Please consider attaching the entire file.

**Comment by 2019-05-10** [comment]
> Hello, @2019-05-10! Please, send the [log files](https://github.com/dbeaver/dbeaver/wiki/Log-files).
Stopped DBeaver.
Started DBeaver.
Experienced severe lags and UI lookup for several minutes and that's what's in the logs:

.log
```
!SESSION 2022-07-14 15:02:02.497 -----------------------------------------------
eclipse.buildId=unknown
java.version=18.0.1-ea
java.vendor=Debian
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_DK
Command-line arguments:  -os linux -ws gtk -arch x86_64

!ENTRY org.jkiss.dbeaver.model 4 0 2022-07-14 15:02:04.582
!MESSAGE No CSSEngine
```

dbeaver-debug.log
```
2022-07-14 15:02:03.468 - Create display
> Start DBeaver UI Editors - Base [org.jkiss.dbeaver.ui.editors.base 1.0.85.202207091909]
2022-07-14 15:02:03.545 - Starting RMI server at 61461
2022-07-14 15:02:03.567 - Run workbench
> Start Eclipse e4 Workbench SWT [org.eclipse.e4.ui.workbench.swt 0.16.500.v20220506-1520]
> Start Eclipse Application Services [org.eclipse.e4.core.services 2.3.200.v20220513-1235]
> Start Eclipse UI Application Services [org.eclipse.e4.ui.services 1.5.0.v20210115-1333]
> Start EMF XML/XMI Persistence [org.eclipse.emf.ecore.xmi 2.16.0.v20190528-0725]
> Start Eclipse CSS SWT Theme Support [org.eclipse.e4.ui.css.swt.theme 0.13.100.v20220310-1056]
> Start Eclipse e4 Workbench Add-ons [org.eclipse.e4.ui.workbench.addons.swt 1.4.400.v20211102-0453]
> Start Eclipse e4 Workbench SWT Renderer [org.eclipse.e4.ui.workbench.renderers.swt 0.15.500.v20220511-1638]
> Start Eclipse Bindings Support [org.eclipse.e4.ui.bindings 0.13.100.v20210722-1426]
> Start Eclipse Dependency Injection Extensions [org.eclipse.e4.core.di.extensions 0.17.100.v20210910-0640]
> Start Eclipse CSS SWT Support [org.eclipse.e4.ui.css.swt 0.14.500.v20220511-1639]
> Start DBeaver UI [org.jkiss.dbeaver.ui 5.1.100.202207091909]
2022-07-14 15:02:04.310 - Configure workbench window
> Start Help System Core [org.eclipse.help 3.9.100.v20210721-0601]
2022-07-14 15:02:04.385 - Create actions
2022-07-14 15:02:04.388 - Disable Eclipse action set 'org.eclipse.ui.edit.text.actionSet.annotationNavigation'
2022-07-14 15:02:04.388 - Disable Eclipse action set 'org.eclipse.ui.WorkingSetActionSet'
2022-07-14 15:02:04.388 - Disable Eclipse action set 'org.eclipse.ui.actionSet.openFiles'
2022-07-14 15:02:04.577 - Initialize workbench window
2022-07-14 15:02:04.581 - Finish initialization
2022-07-14 15:02:04.581 - No CSSEngine
> Start Default Text Editor [org.eclipse.ui.editors 3.14.300.v20210913-0815]
> Start Text Editor Framework [org.eclipse.ui.workbench.texteditor 3.16.500.v20220331-0848]
> Start Equinox Java Authentication and Authorization Service (JAAS) [org.eclipse.equinox.security 1.3.900.v20220108-1321]
> Start Internet Connection Management UI [org.eclipse.ui.net 1.4.0.v20210426-0838]
> Start Internet Connection Management [org.eclipse.core.net 1.3.1200.v20220312-1450]
> Start DBeaver UI Editors - Data [org.jkiss.dbeaver.ui.editors.data 1.0.85.202207091909]
> Start DBeaver SQL Model [org.jkiss.dbeaver.model.sql 1.0.75.202207091909]
2022-07-14 15:02:05.114 - Loading user drivers configuration from '/home/username/.local/share/.DBeaverData/workspace6/.metadata/.plugins/org.jkiss.dbeaver.core/drivers.xml'
2022-07-14 15:02:05.121 - Total database drivers: 99 (99)
> Start DBeaver Sample Database [org.jkiss.dbeaver.ext.sample.database 1.0.122.202207091909]
> Start DBeaver Tip of the day [org.jkiss.dbeaver.ext.ui.tipoftheday 1.0.91.202207091909]
> Start DBeaver UI Editors - SQL [org.jkiss.dbeaver.ui.editors.sql 1.0.85.202207091909]
> Start Draw2d [org.eclipse.draw2dl 3.12.100.20220506142945]
> Start GEF3 GEF (MVC) [org.eclipse.gef3 3.12.100.20220506142945]
> Start DBeaver ERD UI [org.jkiss.dbeaver.erd.ui 3.0.45.202207091909]
> Start DBeaver ERD Model [org.jkiss.dbeaver.erd.model 1.0.45.202207091909]
2022-07-14 15:02:05.234 - Connect with 'PostgreSQL - postgres' (postgres-jdbc-168be124857-7e15cce29d1ec562)
> Start DBeaver PostgreSQL Support [org.jkiss.dbeaver.ext.postgresql 2.1.158.202207091909]
> Start DBeaver GIS Support [org.jkiss.dbeaver.data.gis 1.0.109.202207091909]
2022-07-14 15:02:05.443 - Execution context opened (PostgreSQL - postgres; Main <postgres>; 0)
2022-07-14 15:02:05.443 - Initiate connection to PostgreSQL database [postgres@remotehost] for Main <postgres>
2022-07-14 15:02:05.534 - Connected (postgres-jdbc-168be124857-7e15cce29d1ec562, PostgreSQL JDBC Driver [42.2.5])
2022-07-14 15:02:06.558 - Execution context opened (PostgreSQL - postgres; Main <db1>; 1)
2022-07-14 15:02:06.558 - Initiate connection to PostgreSQL database [db1@remotehost] for Main <db1>
> Start DBeaver PostgreSQL Support UI [org.jkiss.dbeaver.ext.postgresql.ui 1.0.61.202207091909]
2022-07-14 15:02:10.747 - Connect with 'host1:port - db2' (postgres-jdbc-17e6ceb899a-19934618c59ca5f1)
2022-07-14 15:02:10.877 - Execution context opened (host1:port - db2; Main <db2>; 2)
2022-07-14 15:02:10.877 - Initiate connection to PostgreSQL database [db2@host1] for Main <db2>
2022-07-14 15:02:11.069 - Execution context opened (host1:port - db2; Metadata <db2>; 3)
2022-07-14 15:02:11.069 - Initiate connection to PostgreSQL database [db2@host1] for Metadata <db2>
2022-07-14 15:02:11.325 - Connected (postgres-jdbc-17e6ceb899a-19934618c59ca5f1, PostgreSQL JDBC Driver [42.2.5])
> Start Core Compare Support [org.eclipse.compare.core 3.7.0.v20220513-0551]
> Start Console [org.eclipse.ui.console 3.11.200.v20220324-0630]
> Start Debug Core [org.eclipse.debug.core 3.19.100.v20220324-0630]
2022-07-14 15:04:25.832 - Execution context opened (host1:port - db2; SQLEditor <Script-2.sql>; 4)
2022-07-14 15:04:25.832 - Initiate connection to PostgreSQL database [db2@host1] for SQLEditor <Script-2.sql>
2022-07-14 15:04:25.910 - Connect with 'host2 - CON R/O' (postgres-jdbc-16d3a8e50a8-42b23db5505f0b2d)
2022-07-14 15:04:26.200 - Execution context opened (host2 - CON R/O; Main <db3>; 5)
2022-07-14 15:04:26.200 - Initiate connection to PostgreSQL database [db3@host2] for Main <db3>
2022-07-14 15:04:26.664 - Connected (postgres-jdbc-16d3a8e50a8-42b23db5505f0b2d, PostgreSQL JDBC Driver [42.2.5])
2022-07-14 15:04:26.666 - Execution context opened (host2 - CON R/O; SQLEditor <Script-5.sql>; 6)
2022-07-14 15:04:26.666 - Initiate connection to PostgreSQL database [db3@host2] for SQLEditor <Script-5.sql>
```

**Comment by flambchap** [comment]
~@2019-05-10, what kind of packaging do you use?~ Figured out it's a .deb package

Can you please make a thread dump when you experience the issue? https://github.com/dbeaver/dbeaver/wiki/Making-a-thread-dump

**Comment by 2019-05-10** [comment]
> ~@2019-05-10, what kind of packaging do you use?~ Figured out it's a .deb package
> 
> Can you please make a thread dump when you experience the issue? https://github.com/dbeaver/dbeaver/wiki/Making-a-thread-dump

```
$ jstack $(ps aux | grep -m1 dbeaver | awk '{print $2}') > thread-dump.txt
3050041: Unable to open socket file /proc/3050041/root/tmp/.java_pid3050041: target process 3050041 doesn't respond within 10500ms or HotSpotVM not loaded
```

**Comment by 2019-05-10** [comment]
Well, that command is not quite right -- it tries to acces the native launcher process, not the actual JVM process.
This one gets the correct pid:
`jstack $(ps aux | grep java | grep dbeaver | awk '{print $2}') > thread-dump.txt`

[thread-dumps.tar.gz](https://github.com/dbeaver/dbeaver/files/9112993/thread-dumps.tar.gz)

**Comment by ShadelessFox** [comment]
Thanks for the thread dump. This issue is caused by #16887.

**Comment by 2019-05-10** [comment]
I see.
Is there an ETA for an update, or a workaround until then?

**Comment by ShadelessFox** [comment]
I can't think of any workaround, unfortunately.

We've scheduled this issue for the next milestone.

**Comment by Mushoz** [comment]
I had to downgrade due to this issue. A simple Control + Shift + Alt + A to not limit a result to 200 rows on a ~16k row table with a SELECT * FROM table takes less than a second on 22.0.4, but completely hangs the application with 22.1.2. It's completely unusable right now to be frank.

**Comment by jbrains** [comment]
Downgrading to 22.1.1 was enough for me to get back to a comfortable pace of work, in case you prefer not to downgrade all the way to 22.0.x.

**Comment by jbrains** [comment]
I'm merely confirming that this problem is fixed in 22.1.3.
